### PR TITLE
Tenant qualify /identity endpoint

### PR DESCRIPTION
**Pull request by emswbandara**
Addresses https://github.com/wso2/product-is/issues/8626

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java [hunk 1]
```diff
@@ -100,6 +105,25 @@ public void init(InitConfig initConfig) {
...
+    /**
+     * Get tenant qualified callback path.
+     *
+     * @param context IdentityMessageContext
+     * @return Tenant qualified callback path
+     */
+    protected String getTenantQualifiedCallbackPath(IdentityMessageContext context) {
+
+        try {
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && !(new URI(getCallbackPath(context)).isAbsolute()
```
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java:117]: Better to move `(new URI(getCallbackPath(context)).isAbsolute())` to a private method

Comment (mefarazath) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java:117]: Let's move `!getCallbackPath(context).startsWith("/t/")` to a private method

Comment (mefarazath) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java:117]: We need to handle a few cases here 

- tenant qualified mode with absolute callback path
- tenant qualified mode with relative callback path without /t/
- tenant qualified mode with relative callback path with /t/ (already tenant qualified)

- non-tenant qualified mode with absolute callback path
- non-tenant qualified mode with relative callback path without /t/
- non-tenant qualified mode with relative callback path with /t/ (already tenant qualified)

Can we have a unit test to cover these cases?

Comment (emswbandara) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java]: created https://github.com/wso2/product-is/issues/9070 to track this.

**Review by ashensw — APPROVED** [review]

**Pull request merged**